### PR TITLE
add Roxot adapter

### PR DESCRIPTION
**Pull request by kir-roxot**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    "publisherId": "{publisher_id}"
  }
}
```
- contact email : it@roxot.com
- [x] official adapter submission

### CODE REVIEW

**Comment by kir-roxot** [comment]
@protonate @matthewlane Hi guys,
I'v added publisher's account for a test, you can use this config for any site:
```
{
  bidder: 'roxot',
  params: {
    "publisherId": "bea17352-fb94-490d-b206-45096c96dbbc"
  }
}
```
Available sizes: 
```
300x250
728x90
300x600
160x600
320x50
468x60
120x600
250x250
```  
In response you will see ads with our logo.

**Comment by kir-roxot** [comment]
@matthewlane Yes it is, this line is necessary to gather more accurate statistic.

File: src/adapters/roxot.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+var bidfactory = require('../bidfactory.js');
+var bidmanager = require('../bidmanager.js');
+var adloader = require('../adloader');
+
+
+var RoxotAdapter = function RoxotAdapter() {
+  var roxotUrl = "r.rxthdr.com";
+
+  $$PREBID_GLOBAL$$.roxotResponseHandler = roxotResponseHandler;
+  $$PREBID_GLOBAL$$.roxotEvents = events;
```
Comment (mkendall07) [inline src/adapters/roxot.js]: @kir-roxot 
We don't allow exposing internal eventing externally like this. You will need to pass the data through your response handler or use the analytics adapter method to collect data.

**Comment by kir-roxot** [comment]
@matthewlane  @mkendall07 Guys, I had removed expose internal prebid events from roxot adapter, and added roxot analytic adapter like you advised

**Comment by matthewlane** [comment]
The analytics adapter might not be what you want, as it will only work if the publisher explicitly turns this on. Additionally there are `Access-Control-Allow-Origin` errors, which may mean the analytics endpoint isn't configured with CORS support. You may want to try putting the logic for this back in your base adapter rather than an analytics adapter, just take care not to expose any Prebid internals

**Comment by kir-roxot** [comment]
Thank you, for point out the error in analytics endpoint, we added CORS support. Yes we know about turning on analytics collecting, this is fine for us

**Comment by kir-roxot** [comment]
@matthewlane @mkendall07 Guys we really want to be included in upcoming release 0.14, so could you please confirm that's everything is ok, or notify us if we need to do some more changes?

File: src/adapters/roxot.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    var roxotBidReqs = {
+      id: utils.getUniqueIdentifierStr(),
+      bids: bidReqs,
+      site: {
+        domain: domain,
+        page: page
+      }
+    };
+
+    var scriptUrl = '//' + roxotUrl + '?callback=window.pbjs.roxotResponseHandler' +
```
Comment (mkendall07) [inline src/adapters/roxot.js]: you need to use `$$PREBID_GLOBAL$$` instead of `pbjs`

File: src/adapters/roxot.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      '&src=' + CONSTANTS.REPO_AND_VERSION +
+      '&br=' + encodeURIComponent(JSON.stringify(roxotBidReqs));
+
+    adloader.loadScript(scriptUrl, null);
```
Comment (mkendall07) [inline src/adapters/roxot.js]: Don't need null here.

File: src/adapters/roxot.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  function roxotResponseHandler(roxotResponseObject) {
+    utils.logInfo('roxotResponseHandler invoking');
+    var placements = [];
+
+    if (isResponseInvalid()) {
+      return fillPlacementEmptyBid();
+    }
+
+    roxotResponseObject.bids.forEach(pushRoxotBid);
+
```
Comment (mkendall07) [inline src/adapters/roxot.js]: can remove these extra line breaks

**Comment by kir-roxot** [comment]
@mkendall07 @matthewlane I had fixed all remarks  that was in your code review. Could you please review it again?

**Comment by matthewlane** [comment]
Thanks for the update, bids come back and tests are good. Merged

**Pull request merged**